### PR TITLE
MLDB-1313 delay resolution of wildcard ambiguity from parsing to binding

### PR DESCRIPTION
**Pull request by steadtler**
If you take SELECT x.*

instead of assuming it means all column from dataset x, we will wait until binding to see if it means that or all columns that starts with "x."

Add a new method to the SqlBindingScope interface to resolve table names in identifier.

### CODE REVIEW

File: server/dataset_context.h [hunk 1]
```diff
@@ -109,12 +109,16 @@ struct SqlExpressionDatasetContext: public SqlExpressionMldbContext {
     {
         return RowContext(row, params);
     }
+
+    virtual Utf8String 
+    doResolveTableName(const Utf8String & fullVariableName, Utf8String &tableName) const;
```
Comment (guyd) [inline server/dataset_context.h]: It seems that the naming of the function is incorrect.  Are we trying to resolve a variable name or a table name?

File: sql/sql_expression.cc [hunk 2]
```diff
@@ -2019,65 +2013,6 @@ parse(ML::Parse_Context & context, bool allowUtf8)
         }
     }
 
-    //cerr << "offset = " << context.get_offset() << endl;
-    /* Three possibilities:
-       1.  (tablename).(prefix)*
-       2.  expression (AS label)
-       3.  label : expression
-    */
-    if (!matched) {
```
Comment (guyd) [inline sql/sql_expression.cc]: What's happening with that segment of code?  It was never executed?

File: sql/sql_expression_operations.cc [hunk 3]
```diff
@@ -2330,18 +2330,24 @@ WildcardExpression(Utf8String tableName,
...
       excluding(excluding)
 {
+
 }
 
 BoundSqlExpression
 WildcardExpression::
 bind(SqlBindingScope & context) const
 {
+    Utf8String simplifiedprefix = prefix;
```
Comment (guyd) [inline sql/sql_expression_operations.cc]: simplifiedprefix -> simplifiedPrefix

File: server/dataset_context.h (hunk 1 above)
Comment (steadtler) [inline server/dataset_context.h]: Well, both as it were. We have a variable name that may or may not contain a table name.

File: sql/sql_expression.cc (hunk 2 above)
Comment (steadtler) [inline sql/sql_expression.cc]: That code was there to find the explicit dataset.* case before we handle the more generic case identifier*  Now we can't handle the explicit case at parsing time, so the code isnt needed anymore.

**Comment by jeremybarnes** [comment]
I agree very strongly with that change.  I'm surprised it was such a small amount of code to change.  +1 from me.

**Pull request merged**